tests/lib/rules/newline-per-chained-call.js

```diff
@@ -0,0 +1,114 @@
+/**
+ * @fileoverview Tests for newline-per-chained-call rule.
+ * @author Rajendra Patil
+ * @copyright 2016 Rajendra Patil. All rights reserved.
+ */
+
+"use strict";
+
+//------------------------------------------------------------------------------
+// Requirements
+//------------------------------------------------------------------------------
+
+var rule = require("../../../lib/rules/newline-per-chained-call"),
+    RuleTester = require("../../../lib/testers/rule-tester");
+
+var ruleTester = new RuleTester();
+ruleTester.run("newline-per-chained-call", rule, {
+    valid: [{
+        code: "_\n.chain({})\n.map(foo)\n.filter(bar)\n.value();"
+    }, {
+        code: "a.b.c.d.e.f"
+    }, {
+        code: "a()\n.b()\n.c\n.e"
+    }, {
+        code: "var a = m1.m2(); var b = m1.m2();\nvar c = m1.m2()"
+    }, {
+        code: "var a = m1()\n.m2();"
+    }, {
+        code: "var a = m1();"
+    }, {
+        code: "var a = m1().m2.m3();",
+        options: [{
+            ignoreChainWithDepth: 3
+        }]
+    }, {
+        code: "var a = m1().m2.m3().m4.m5().m6.m7().m8;",
+        options: [{
+            ignoreChainWithDepth: 8
+        }]
+    }],
+    invalid: [{
+        code: "_\n.chain({}).map(foo).filter(bar).value();",
+        errors: [{
+            message: "Expected line break after `_\n.chain({}).map(foo).filter(bar)`."
```

Comment:
@platinumazure I'd prefer to keep that decision local for now, as just removing newlines might not be the right approach. For instance, in this case, you could also choose to change the message to just use ".filter(bar)".